src/bin/pg_autoctl/pgctl.c

```diff
@@ -252,6 +252,67 @@ config_find_pg_ctl(PostgresSetup *pgSetup)
 }
 
 
+/*
+ * find_extension_control_file ensures that the extension is present in the
+ * given Postgres installation. This does the equivalent of:
+ * ls -l $(pg_config --sharedir)/extension/pg_stat_statements.control
+ */
+bool
+find_extension_control_file(PostgresSetup *pgSetup, char *extName)
```

Comment:
I would still like it better if it took a `const char *pg_ctl` argument rather than the whole PostgresSetup, singling the read-only aspect of the operation in terms of in-memory data structures...

Reply:
Sorry! I totally missed your earlier comment